test/coap_message_api_test.dart

```diff
@@ -242,77 +242,109 @@ void main() {
   test('Uri path', () {
     final message = CoapMessage();
     expect(message.uriPaths.length, 0);
-    message.uriPath = 'a/uri/path/';
-    expect(message.uriPaths.length, 3);
-    expect(message.uriPathsString, 'a/uri/path');
+    for (final path in ['/a/uri/path', 'a/uri/path/', '/a/uri/path/']) {
+      message.uriPath = path;
+      expect(message.uriPaths.length, 3);
+      expect(message.uriPath, 'a/uri/path');
+    }
     message.addUriPath('longer');
     expect(message.uriPaths.length, 4);
-    expect(message.uriPathsString, 'a/uri/path/longer');
-    expect(() => message.addUriPath(null), throwsArgumentError);
+    expect(message.uriPath, 'a/uri/path/longer');
+    expect(
+      () => message.addUriPath('multiple/not/allowed'),
+      throwsArgumentError,
+    );
+    expect(
+      () => message.addLocationPath('no-double-slash//'),
```

Comment:
Technically I think an URI path option can be empty, so `a//b/c` might be a valid URI. Not sure how to feel about that, the reason I added this check is I didn't want people adding multiple paths using `addLocationPath` (since it's intended for adding a single path option or path segment).